### PR TITLE
YD-300 Corrected the week number

### PR DESCRIPTION
**Pull request by Bert-R**

### CODE REVIEW

File: appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy [hunk 1]
```diff
@@ -544,6 +551,20 @@ class ActivityTest extends AbstractAppServiceIntegrationTest
...
+		boolean isSunday = YonaServer.now.getDayOfWeek() == DayOfWeek.SUNDAY
+		DateTimeFormatter weekFormatter = new DateTimeFormatterBuilder()
+				.parseCaseInsensitive()
+				.appendValue(IsoFields.WEEK_BASED_YEAR, 4, 10, SignStyle.EXCEEDS_PAD)
+				.appendLiteral("-W")
+				.appendValue(IsoFields.WEEK_OF_WEEK_BASED_YEAR, 2)
+				.toFormatter(YonaServer.EN_US_LOCALE)
+
+		String currentWeek = weekFormatter.format(YonaServer.now)
+		assert responseWeekOverviews.responseData._embedded."yona:weekActivityOverviews"[0].date == currentWeek || isSunday
```
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy]: Maybe add a comment why it doesn't work on Sunday.

File: core/src/main/java/nu/yona/server/analysis/service/IntervalActivityOverviewDTO.java [hunk 2]
```diff
@@ -34,9 +32,11 @@ public String getTimeZoneId()
...
-	 * The ISO-8601 date formatter.
+	/**
+	 * Format the date in compliance with ISO-8601
+	 * 
+	 * @param localDate The date to be formatted
+	 * @return The formatted date
 	 */
-	@JsonIgnore
-	protected abstract DateTimeFormatter getDateFormatter();
+	protected abstract String formatDateAsISO(LocalDate localDate);
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/analysis/service/IntervalActivityOverviewDTO.java]: Should the `@JsonIgnore` not be retained here?

Comment (aristotelos) [inline core/src/main/java/nu/yona/server/analysis/service/IntervalActivityOverviewDTO.java]: Sorry, my mistake, it is not a getter any more.

**Comment by aristotelos** [comment]
Looks great, added one comment.

File: appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy (hunk 1 above)
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy]: Done.

**Pull request merged**